enterprise/internal/scim/user.go

```diff
@@ -39,16 +41,117 @@ func NewUserResourceHandler(ctx context.Context, observationCtx *observation.Con
 }
 
 // Create stores given attributes. Returns a resource with the attributes that are stored and a (new) unique identifier.
-func (h *UserResourceHandler) Create(r *http.Request, attributes scim.ResourceAttributes) (scim.Resource, error) {
-	// TODO: For testing only, real logic should come here
-	attributesString := resourceAttributesToLoggableString(attributes)
-	h.observationCtx.Logger.Error("XXXXX Create", log.String("method", r.Method), log.String("attributes", attributesString))
+func (h *UserResourceHandler) Create(_ *http.Request, attributes scim.ResourceAttributes) (scim.Resource, error) {
+	// Get external ID
+	optionalExternalID := getOptionalExternalID(attributes)
+
+	// Get primary email: try to get the (first) email address marked as primary, otherwise use the first email address
+	primaryEmail := ""
+	if attributes["emails"] == nil {
+		return scim.Resource{}, scimerrors.ScimErrorBadParams([]string{"emails missing"})
+	}
+	emails := attributes["emails"].([]interface{})
+	for _, email := range emails {
+		if email.(map[string]interface{})["primary"] == true {
+			primaryEmail = email.(map[string]interface{})["value"].(string)
+			break
+		}
+	}
+	if primaryEmail == "" && len(emails) > 0 {
+		primaryEmail = emails[0].(map[string]interface{})["value"].(string)
+	}
+
+	// Get username
+	username := ""
+	if attributes["userName"] != nil {
+		username = attributes["userName"].(string)
+	}
+
+	// Get displayName
+	displayName := ""
+	if attributes["displayName"] != nil {
+		displayName = attributes["displayName"].(string)
+	} else if attributes["name"] != nil {
+		displayName = attributes["name"].(map[string]interface{})["formatted"].(string)
+		if displayName == "" && attributes["name"].(map[string]interface{})["givenName"] != nil && attributes["name"].(map[string]interface{})["familyName"] != nil {
+			displayName = attributes["name"].(map[string]interface{})["givenName"].(string) + " " + attributes["name"].(map[string]interface{})["familyName"].(string)
+		}
+	}
+
+	// Create user (with or without external ID)
+	// TODO: Use NewSCIMUser instead of NewUser?
+	newUser := database.NewUser{
+		Email:           primaryEmail,
+		Username:        username,
+		DisplayName:     displayName,
+		EmailIsVerified: true,
+	}
+	var user *types.User
+	var err error
+	if optionalExternalID.Present() {
+		accountSpec := extsvc.AccountSpec{
+			ServiceType: "scim",
+			ServiceID:   "TODO",
```

Comment:
I think it might be better to add a TODO comment not to forget about this:
```suggestion
			// TODO: provide proper service ID
			ServiceID:   "TODO",
```